### PR TITLE
Api token v2/fix sync actions

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

- Sync actions in content API boostrap instead of in syncPermissions
- add `contentAPI.permissions.getActionsMap()` that provides an object of all actions in a format for the front-end

### Why is it needed?

It's where it should be done, and it's necessary to provide the response for the front-end to build the segregated actions list

### How to test it?

Make sure custom tokens with permissions ca be created without getting any error about action not being registered.

Tests have also been added

### CODE REVIEW

File: packages/core/strapi/lib/services/content-api/index.js [hunk 1]
```diff
@@ -2,20 +2,36 @@
...
-     * NOTE: For some reason, this doesn't seem to be necessary because all the routes exist
-     * createActionProvider uses a providerFactory, which seems to already include everything, and when we try
-     * to register our actions we get an error that the keys already exist
-     * Could providerFactory not be providing a new provider, and instead sharing the registry with everything that uses it?
-     *
-     * If this isn't an issue to fix and is expected, we don't need the route registration code below and it should be removed
-     * */
-    // Start of route registration
+    // Register actions
     const apiRoutesName = Object.values(strapi.api)
```
Comment (Convly) [inline packages/core/strapi/lib/services/content-api/index.js:35]: I know this is what I initially pushed to create all the actions UID but I'm curious about whether we should use routes handler name (this) or controllers name (what we use in U&P for now).

The benefit I see by doing this approach is that we're not leaking controllers that are not associated with a route. But the downside is that if a controller is linked to a route dynamically, we'll miss it.

@alexandrebodin any opinion on this?

File: packages/core/strapi/lib/services/content-api/index.js [hunk 2]
```diff
@@ -34,18 +50,11 @@ const createContentAPI = (/* strapi */) => {
     Promise.all(
       uniq(actions).map((action) =>
         providers.action.register(action).catch(() => {
-          // console.log('Key already exists', action);
+          console.warn('Trying to add action that already exists', action);
```
Comment (Convly) [inline packages/core/strapi/lib/services/content-api/index.js:53]: Maybe we could use `strapi.log.warn` here to benefit from the user's Winston configuration, wdyt?

**Review by Convly — COMMENTED** [review]
Nice, some questions :slightly_smiling_face:

File: packages/core/strapi/lib/services/content-api/index.js (hunk 1 above)
Comment (Convly) [reply packages/core/strapi/lib/services/content-api/index.js:35]: ref: 

https://github.com/strapi/strapi/blob/main/packages/plugins/users-permissions/server/services/users-permissions.js#L154-L170

https://github.com/strapi/strapi/blob/main/packages/plugins/users-permissions/server/services/users-permissions.js#L32-L98

File: packages/core/strapi/lib/services/content-api/index.js (hunk 2 above)
Comment (innerdvations) [reply packages/core/strapi/lib/services/content-api/index.js:53]: Yes, I didn't know that existed, thanks 😆

**Pull request merged**